### DISCUSSION TITLE
Calimar Pump Control via Relays

### DISCUSSION THREAD

**Discussion by dougcfl** [tagyoureit/nodejs-poolController#828] [Q&A] [answered]
Hi. I desire to control a Calimar pump (with the Calimar automation controller installed on the pump) and I'm having an issue with the relays connected to REM staying engaged.  I know my relay board works as the pool/spa lights and valve relays work as expected.  However when activating the pump relays, they toggle on/off repeatedly with a few seconds between the cycle.  I don't see any errors in the log.  I configured a two speed pump in Nixie, but actually there are four relays that can control this pump.  I'd appreciate some help figuring out how to configure Nixie to latch these relays when in Pool or SPA mode.  What should I look at for additional troubleshooting?
Thanks.

**Comment by rstrouse** [comment]
How many speeds does the pump have?  I know it has 4 relays but how many speeds?  This will let me know whether they used the SuperFlo or the HW model on the controller.

EDIT:  Is it compatible with this?
https://www.poolsupplyunlimited.com/pool/calimar-cmarvspibpump-interface-board/187101p1

**Reply by dougcfl** [reply]
Yep. That’s the pump controller. The pump has four set-able speeds. Speeds 1-3 can be controlled with the relays. The fourth relay starts a quick clean cycle. My understanding is speed 4 is not accessible with a relay. I learned about this pump (and nodejs-poolController) on TroubleFreePool. I don’t yet have the pump. It’s in transit to me now.

**Reply by rstrouse** [reply]
If that is the pump controller then you can connect it to RS485 and define it as an IntelliFlo VS.  It will then allow you to set any speed between 450rpm to 3450rpm.

EDIT:  Hooked up like that you will not be using relays.

**Reply by dougcfl** [reply]
That would be even better.  How do I wire it to the RS485 bus?  I see instructions for wiring only to relays using the connections 5V, IN1, IN2, IN3, IN4.  And two available schemes for control.  I don't see any serial connections in the installation manual.  https://www.poolsupplyunlimited.com/Products/Manuals/187101_2022112314226.pdf

Here is the actual pump, no serial connection there either.  https://www.poolsupplyunlimited.com/pool/calimar-cmarvsp30-variable-speed-pool-pump/168195p1

**Reply by dougcfl** [reply]
Here’s the connections for the automation interface. Are the A and B connections for serial rs485?  I’ve done MIDI serial but not rs485. Don’t I need data+, data-, and ground for rs485?

**Reply by Kaiyotech** [reply]
> If that is the pump controller then you can connect it to RS485 and define it as an IntelliFlo VS. It will then allow you to set any speed between 450rpm to 3450rpm.
> 
> EDIT: Hooked up like that you will not be using relays.

@rstrouse have you actually used this controller over RS-485, or how did you know this? The manual doesn't mention it at all. I'm considering trying it, but I'm about to buy all my stuff, and I need an extra relay board HAT if this won't work over Rs-485, so I kinda need to know if this has any chance.

The manual I'm just getting from here: https://www.poolsupplyunlimited.com/pool/calimar-cmarvspib-pump-interface-board/187101p1

**Comment by dougcfl** [comment]
![IMG_2115](https://github.com/tagyoureit/nodejs-poolController/assets/47368669/b455d1b8-54ca-4c9d-834e-6d5e52a55949)

**Reply by rstrouse** [reply, marked as answer]
No RS485 is a differential voltage signal.  You only need A/B.

**Reply by tagyoureit** [reply]
The manuals don't mention RS485 compatibility.  Will be curious to see if it works.  Keep us posted.

**Reply by Kaiyotech** [reply]
@dougcfl Did this end up working? Any issues at all? You just wire the A/B on the pump controller (that's what's in the picture?) to a simple RS485 hat on your Pi I assume (like this? https://www.pishop.us/product/rs485-can-hat-for-raspberry-pi/#:~:text=Waveshare%2014882%20%2D%20RS485%20CAN%20HAT,work%20with%20other%20control%20boards).

I'm very interested because this saves me an extra relay hat plus just much more fun to control rpm directly.

**Reply by dougcfl** [reply]
I didn't try it.  I ended up using some amazon relays and routines to handle my automation.
